javaparser-core/src/main/java/com/github/javaparser/ast/body/BodyDeclaration.java

```diff
@@ -282,4 +283,48 @@ public void ifMethodDeclaration(Consumer<MethodDeclaration> action) {
     @Generated("com.github.javaparser.generator.core.node.TypeCastingGenerator")
     public void ifTypeDeclaration(Consumer<TypeDeclaration> action) {
     }
+
+    public Optional<AnnotationDeclaration> toAnnotationDeclaration() {
```

Comment:
In practice I think I would prefer to get directly the type, without the Optional wrapping it. Otherwise my code would start having a ton of `get()`

Reply:
I've been looking into JSS Navigator a little, and my advice is to look into map and ifPresent a lot more, since those two replace many use cases for the current ifXyz/asXyz. This PR doesn't affect your code by the way :-)